### PR TITLE
Remove support for skip_rows / num_rows options in the parquet reader.

### PR DESCRIPTION
**Pull request by nvdbaranec**
Removes support for skip_rows / num_rows options in the parquet reader.  Users retain control of what gets read via row groups.

Did some before/after benchmarking.  As expected, this doesn't change much except for a minor boost in list reading (due to simplification of the preprocessing step).  Most of the ways the row bounds affected the code was in the page setup process (making it slippery to think through the logic) and didn't do much in the actual process of decoding.  A selection of before/after benchmarks (all input files ~512  MB)

```
ParquetRead/integral_buffer_input/29/1000/32/0/1/manual_time
Before:  bytes_per_second=31.4564G/s
After:   bytes_per_second=31.58G/s

ParquetRead/floats_buffer_input/31/1000/32/0/1/manual_time
Before:  bytes_per_second=49.2819G/s
After:   bytes_per_second=49.7408G/s

ParquetRead/string_file_input/23/1000/32/0/0/manual_time
Before:  bytes_per_second=24.634G/s
After:   bytes_per_second=24.6563G/s

ParquetRead/string_buffer_input/23/0/1/0/1/manual_time
Before:  bytes_per_second=5.03313G/s
After:   bytes_per_second=5.03535G/s

ParquetRead/list_buffer_input/24/0/1/1/1/manual_time
Before:  bytes_per_second=1.11488G/s
After:   bytes_per_second=1.31447G/s
```

### CODE REVIEW

**Comment by vuule** [comment]
Please post the impact on the reader benchmarks.

**Review by hyperbolic2346 — APPROVED** [review]
I like these kinds of changes!

File: python/cudf/cudf/_lib/parquet.pyx [hunk 1]
```diff
@@ -151,7 +151,6 @@ cpdef read_parquet(filepaths_or_buffers, columns=None, row_groups=None,
 
     cdef bool cpp_strings_to_categorical = strings_to_categorical
     cdef bool cpp_use_pandas_metadata = use_pandas_metadata
-
```
Comment (PointKernel) [inline python/cudf/cudf/_lib/parquet.pyx:154]: We could revert this file if changes are not necessary.

**Review by PointKernel — APPROVED** [review]
LGTM

Python binding changes seem a cleanup leftover.

File: python/cudf/cudf/_lib/parquet.pyx (hunk 1 above)
Comment (nvdbaranec) [reply python/cudf/cudf/_lib/parquet.pyx]: Done

**Comment by nvdbaranec** [comment]
Just realized the CI failure was because of actual compile errors in the benchmarks.  Fixed.

**Review by hyperbolic2346 — APPROVED** [review]

**Review by PointKernel — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**